packages/flutter/lib/src/widgets/scroll_activity.dart

```diff
@@ -149,6 +149,56 @@ class IdleScrollActivity extends ScrollActivity {
   bool get isScrolling => false;
 }
 
+/// Interface for holding a [Scrollable] stationary.
+///
+/// An object that implements this interface is returned by
+/// [ScrollPosition.hold]. It holds the scrollable stationary until an activity
+/// is started or the [cancel] method is called.
+abstract class ScrollHoldController {
+  /// Release the [Scrollable], potentially letting it go ballistic if
+  /// necessary.
+  void cancel();
+}
+
+/// A scroll activity that does nothing but can be released to resume
+/// normal idle behavior.
+///
+/// This is used while the user is touching the [Scrollable] but before the
+/// touch has become a [Drag].
+///
+/// For the purposes of [ScrollNotification]s, this activity does not constitute
+/// scrolling, and does not prevent the user from interacting with the contents
+/// of the [Scrollable] (unlike when a drag has begun or there is a scroll
+/// animation underway).
+class HoldScrollActivity extends ScrollActivity implements ScrollHoldController {
+  /// Creates a scroll activity that does nothing.
+  HoldScrollActivity({
+    @required ScrollActivityDelegate delegate,
+    this.onHoldCanceled,
```

Comment:
For dragging, we put this callback on the controller rather than the activity.  Putting it on the activity looks cleaner.  I wonder if we can do that for dragging too.